src/Microsoft.ServiceFabric.Services.Remoting/Description/InterfaceDescription.cs

```diff
@@ -67,10 +67,13 @@ public MethodDescription[] Methods
             MethodReturnCheck methodReturnCheck)
         {
             EnsureValidMethods(remotedInterfaceKindName, remotedInterfaceType, methodReturnCheck);
-            var methods = remotedInterfaceType.GetMethods().Select(
-                methodInfo => MethodDescription.Create(remotedInterfaceKindName, methodInfo)).ToList();
-
-            return methods.ToArray();
+            var methods = remotedInterfaceType.GetMethods();
```

Comment:
Could you explain what causes perf gain here? Is the perf gain  because Linq .ToArray() is not performant. It would be great if you can share your perf numbers and the perf code as well.